File: LeetCode/1351统计有序矩阵中的负数.java
class Solution {
    public int countNegatives(int[][] grid) {
        int cnt = 0;
        for (int i = 0; i < grid.length; i++) {
            int left = 0, right = grid[i].length - 1;
            while (left <= right) {
                int mid = left + (right - left) / 2;
                if (grid[i][mid] >= 0) left = mid + 1;
                else right = mid - 1;
            }
            cnt += grid[i].length - left; 
        }
        return cnt;
    }
}

File: LeetCode/1351统计有序矩阵中的负数.go
func countNegatives(grid [][]int) int {
	cnt := 0
	for i := 0; i < len(grid); i++ {
		for j := 0; j < len(grid[0]); j++ {
			if grid[i][j] < 0 {
				cnt++
			}
		}
	}
	return cnt
}